### PR TITLE
chore: miscellaneous cleanups in iot tests

### PR DESCRIPTION
**Pull request by jamesonwilliams**
1. Use Java 7 <> diamond operator
  2. Avoid one and two letter identifiers
  3. Fix indentation inconsistencies
  4. Resolve miscellaneous Android Studio lint issues
  5. Remove (mostly useless) test base class
  6. Split configuration classes out to top level

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/ConnectionParams.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private final String customerAuthorizerName;
+
+    ConnectionParams(final String tokenKeyName,
+                     final String token,
+                     final String tokenSignature,
+                     final String customerAuthorizerName) {
+        this.tokenKeyName = tokenKeyName;
+        this.token = token;
+        this.tokenSignature = tokenSignature;
+        this.customerAuthorizerName = customerAuthorizerName;
```
Comment (jpignata) [inline aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/ConnectionParams.java:34]: Is this a typo? (customer -> custom)?

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java [hunk 2]
```diff
@@ -31,88 +30,85 @@
...
-    private static final String TAG = MqttManagerIntegrationTest.class.getSimpleName();
+    @BeforeClass
+    public static void beforeSuite() throws Exception {
+        Context appContext = InstrumentationRegistry.getTargetContext();
+        keyStorePath = appContext.getFilesDir().toString() + "/";
+        System.out.println(keyStorePath);
+        credentialsProvider = new CognitoCachingCredentialsProvider(
+            appContext,
+            getPackageConfigure("iot").getString(ConfigKey.IDENTITY_POOL_ID.toString()),
+            Regions.US_EAST_1
```
Comment (jpignata) [inline aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java:86]: Idle thought: I wonder if we should move all of these region declarations to the `testconfiguration.json`. Maybe someday later.

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/ConnectionParams.java (hunk 1 above)
Comment (jamesonwilliams) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/ConnectionParams.java:34]: Good catch! Will update.

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java (hunk 2 above)
Comment (jamesonwilliams) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java:86]: I like it, I'll go for it.

Comment (jpignata) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java:86]: Probably not worth it but I defer to you!

Comment (jamesonwilliams) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java:86]: Eep, strike that. Ended up being a bigger deal than that. It isn't always us-east-1.

**Pull request merged**